content/articles/api-authentication-with-django-knox-and-postman-testing/index.md

```diff
@@ -0,0 +1,343 @@
+
+Furthermore, the views for the registration and login shall be done inside the file named `api.py` in our `account` app. 
+
+```python
+from django.contrib import auth
+from rest_framework import generics, permissions, serializers
+from rest_framework.response import Response
+from knox.models import AuthToken
+from .serializers import UserSerializer, RegisterSerializer, LoginSerializer
+
+class SignUpAPI(generics.GenericAPIView):
+    serializer_class = RegisterSerializer
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.save()
+        token = AuthToken.objects.create(user)
+        return Response({
+            "users": UserSerializer(user, context=self.get_serializer_context()).data,
+            "token": token[1]
+        })
+
+
+class SignInAPI(generics.GenericAPIView):
+    serializer_class = LoginSerializer
+
+    def post(self, request):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.validated_data
+        return Response({
+            "user": UserSerializer(user, context=self.get_serializer_context()).data,
+            "token": AuthToken.objects.create(user)[1]
+        })
+
+
+class MainUser(generics.RetrieveAPIView):
+  permission_classes = [
+      permissions.IsAuthenticated
+  ]
+  serializer_class = UserSerializer
+
+  def get_object(self):
+    return self.request.user
+```
+
+Here are the explanations for each of the classes.
+
+As for the `RegisterAPI` and `LoginAPI` classes, we are providing the serializers made earlier to the `serializer_class` due to the class-based views that we used.
+
+The method `post` will be triggered whenever a user is try to register by sending data and a token is generated at the same time. For every user that registered successfully, a token will be sent along with the data for login.
+
+The `UserAPI` will make the user available from the `User` model. Also, the authentication will be done on the received user.
+
+Lastly, the `urls.py` file in the `account` app will be used to make URLs endpoints for the registration and login of users. 
+
+We will be importing the APIViews made in `api.py` as the views for the routes. Here is the code for the routing.
+
+```python
+from django.urls import path, include
+from .api import SignUpAPI, SignInAPI, MainUser
+from knox import views as knox_views
+
+urlpatterns = [
+    path('api/auth/', include('knox.urls')),
+    path('api/auth/register', SignUpAPI.as_view()),
+    path('api/auth/login', SignInAPI.as_view()),
+    path('api/auth/user', MainUser.as_view()),
+    path('api/auth/logout',knox_views.LogoutView.as_view(), name="knox-logout"),
+]
+```
+
+Note that we are including `knox.urls`, this is to route every HTTP calls with the endpoints `api/auth` to match the secured API. 
```

Comment:
Change to " Note that we include `knox.urls`. This is to route every HTTP calls with the endpoints `api/auth` to match the secured API."